docs/odh/fetch-upstream-release.md

```diff
@@ -0,0 +1,430 @@
+This repository is a fork of [kserve/kserve](https://github.com/kserve/kserve)
+repository to integrate with Open Data Hub platform. The releases on ODH
+platform have a faster cadence than the upstream repository. Still, an upstream
+stable release is used as a base for any releases of the ODH Project. We
+accomplish this by creating a branch in ODH fork based from an upstream release
+tag. Any integration changes to ODH Platform are added on top of the created
+branch.
+
+This guide outlines the process to fetch a stable upstream release to ODH fork.
+
+# Prerequisites
+
+As you will be are creating a _new_ branch, the configured branch protections
+may not apply to it. Despite this, you must still adhere to [GitHub
+flow](https://docs.github.com/en/get-started/using-github/github-flow).
+
+You will need:
+
+* Push rights to
+  [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/) repository.
+  * This is to allow you doing the initial push of the release branch.
+  * If you don't have push rights, you can ask somebody else to do the initial
+    push.
+* Your own fork of
+  [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/).
+  * This is to stay adhered to [GitHub
+    flow](https://docs.github.com/en/get-started/using-github/github-flow).
+  * You can create a fork on your account using this link:
+    https://github.com/opendatahub-io/kserve/fork.
+  * If you have push rights to
+    [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/) the fork
+    may be perceived as optional. Still, we prefer to keep the repository clean
+    from working branches and open pull requests from contributor forks.
+* Your own fork of
+  [openshift/release](https://github.com/openshift/release) repository.
+  * This is to configure CI for the new release branch.
+  * You can create a fork on your account using this link:
+    https://github.com/openshift/release/fork.
+* CLI tools:
+  * `jq` CLI as a helper tool [\[homepage\]](https://jqlang.org/).
+  * `podman` because some operations are ran inside containers
+  * `make` to run Makefiles
+  * Of course: `git` command.
+  * Other CLI tools: `curl`, `grep`, `sed`, `xdg-open`.
+
+# Branch cut procedure
+
+## Preparation of the working copy
+Clone your fork of [opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/):
+
+```sh
+GH_USER={your-github-user}
+
+# Use one of the following commands:
+
+# If you prefer the remote configured with SSH
```

Comment:
We should always recommend it over https

Reply:
I think GitHub does not recommend one over the other. So, I tried to be neutral here. Is there any reason for recommending SSH?

Reply:
security reasons, mainly.

Reply:
OK, I hope I removed all mentions of HTTPS.